FIX Raise error when n_neighbours >= n_samples / 2  in manifold.trustworthiness (#23033)
scikit-learn/scikit-learn: doc/whats_new/v1.1.rst

  :pr:`21805` by :user:`Jeanne Dionisi <jeannedionisi>` and :pr:`22685` by
  :user:`Meekail Zain <micky774>`.

> - |Enhancement| Add a warning for :func:`manifold.trustworthiness` if
>   `n_neighbours` is less than `n_samples / 2` to ensure a correct
>   support for the function.

Reviewer: ```suggestion
- |Enhancement| :func:`manifold.trustworthiness` raises a warning if
  `n_neighbours >= n_samples / 2` to ensure a correct support for the function.
```

Reviewer 2: It raises an error now
```suggestion
- |Enhancement| :func:`manifold.trustworthiness` raises an error if
  `n_neighbours >= n_samples / 2` to ensure a correct support for the function.
```
